### PR TITLE
Update nlargest nsmallest doc

### LINKED ISSUE #45041: DOC: pd.Series.nlargest(keep='all') does not ignore duplicates when counting the largest values
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
In [1]: import pandas as pd

In [3]: s = pd.Series([1,2,3,3,4])
   ...: s.nlargest(2, keep='all')
Out[3]:
4    4
2    3
3    3
dtype: int64

In [4]: s = pd.Series([1,2,3,3,4])
   ...: s.nlargest(3, keep='all')
Out[4]:
4    4
2    3
3    3
dtype: int64

In [5]: pd.__version__
Out[5]: '1.4.0.dev0+1453.ga3c0e7bcfb
```

### Issue Description

nlargest = 2 and nlargest = 3 gives the same output here with `keep='all'`

### Expected Behavior

I'd expect `s.nlargest(3, keep='all')` to return 
```py
4    4
2    3
3    3
1    2
dtype: int64
```

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : a3c0e7bcfb8bbe9ca45df7e571a305d403e0f066
python           : 3.8.12.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.128-microsoft-standard
Version          : #1 SMP Tue Jun 23 12:58:10 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.0.dev0+1453.ga3c0e7bcfb
numpy            : 1.21.2
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3
setuptools       : 58.2.0
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.23.2
sphinx           : 4.2.0
blosc            : None
feather          : None
xlsxwriter       : 3.0.1
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.2
IPython          : 7.28.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.2
fsspec           : 2021.10.0
fastparquet      : 0.7.1
gcsfs            : 2021.10.0
matplotlib       : 3.4.3
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 5.0.0
pyxlsb           : None
s3fs             : 2021.10.0
scipy            : 1.7.1
sqlalchemy       : 1.4.25
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : 0.18.2
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.53.1
</details>

### PR DESCRIPTION
**Pull request by MainHanzo**
- [x] closes #45041 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looks great! Just one more request.

File: pandas/core/frame.py [hunk 1]
```diff
@@ -7565,6 +7567,16 @@ def nlargest(
         Maldives      434000     4520      MV
         Brunei        434000    12128      BN
 
+        However, ``nlargest`` does not keep ``n`` distinct largest elements:
+        
+        >>> df.nlargest(5, 'population', keep='all')
```
Comment (rhshadrach) [inline pandas/core/frame.py:7572]: This is a great example, can you also include a similar one in the docstring of `nsmallest`?

Comment (MainHanzo) [reply pandas/core/frame.py:7572]: @rhshadrach  done.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
@MainHanzo - pre-commit is failing with `black`. If you aren't familiar with pre-commit, see here:

https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#pre-commit

**Comment by MainHanzo** [comment]
@rhshadrach it’s all right now

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MainHanzo

**Pull request merged**